fe1-web/src/features/evoting/network/__tests__/ElectionHandler.test.ts

```diff
@@ -0,0 +1,201 @@
+import 'jest-extended';
+import '__tests__/utils/matchers';
+import {
+  mockLaoId,
+  mockLaoIdHash,
+  mockLaoName,
+  configureTestFeatures,
+  mockKeyPair,
+  mockLaoState,
+} from '__tests__/utils';
+
+import { Hash, Timestamp, Base64UrlData, Signature } from 'core/objects';
+import { ActionType, MessageData, ObjectType } from 'core/network/jsonrpc/messages';
+
+import STRINGS from 'resources/strings';
+import { getStore, dispatch } from 'core/redux';
+import { connectToLao } from 'features/lao/reducer';
+import { addEvent } from 'features/events/reducer';
+import { getEventFromId } from 'features/events/network/EventHandlerUtils';
+import { Election, ElectionStatus } from 'features/evoting/objects';
+import { handleElectionOpenMessage } from '../ElectionHandler';
+
+const store = getStore();
+
+const TIMESTAMP = new Timestamp(1609455600); // 1st january 2021
+const CLOSE_TIMESTAMP = new Timestamp(1609542000); // 2nd january 2021
+
+const mockElectionId = Hash.fromStringArray(
+  'Election',
+  mockLaoId,
+  TIMESTAMP.toString(),
+  mockLaoName,
+);
+
+const election = new Election({
+  lao: mockLaoIdHash,
+  id: mockElectionId,
+  name: 'An election',
+  version: STRINGS.election_version_identifier,
+  createdAt: TIMESTAMP,
+  start: TIMESTAMP,
+  end: CLOSE_TIMESTAMP,
+  questions: [],
+  electionStatus: ElectionStatus.NOT_STARTED,
+  registeredVotes: [],
+});
+
+const mockMessageData = {
+  receivedAt: TIMESTAMP,
+  laoId: mockLaoIdHash,
+  data: Base64UrlData.encode(''),
+  sender: mockKeyPair.publicKey,
+  signature: Base64UrlData.encode('') as Signature,
+  channel: '',
+  message_id: Hash.fromString(''),
+  witness_signatures: [],
+};
+
+const initializeData = () => {
+  // clear data in the redux store
+  dispatch({ type: 'CLEAR_STORAGE', value: {} });
+};
+
+beforeAll(() => {
+  configureTestFeatures();
+  initializeData();
+});
+
+describe('ElectionHandler', () => {
+  describe('election#open', () => {
+    it('should return false if the object is not "election"', () => {
+      const mockFn = jest.fn();
+      console.warn = mockFn;
```

Comment:
How would I do that? The reason for doing it that way was to be able to check how many times it was called during that test and if I just did it in a `beforeEach` I will get typescript errors since it does not know we overwrote `console.warn` with a mock function. (i.e. `mockFn.mock.calls[0][0]` will not be typechecked correctly)